assets/scripts/configure-microk8s-macos.sh

```diff
@@ -0,0 +1,282 @@
+#!/bin/bash
+
+#==========
+# Functions
+#==========
+
+checkExitCode() {
+    exit_status=$?
+    if [ $exit_status -ne 0 ]; then
+        exit $exit_status
+    fi
+}
+
+#===========
+# Parameters
+#===========
+
+while getopts a:c:d:f:i:o:p:r: flag; do
+    case "${flag}" in
+    a) ASSETS_FOLDER=${OPTARG} ;;
+    c) CONFIGS_FOLDER=${OPTARG} ;;
+    d) FORCE_DB_REFRESH=${OPTARG} ;;
+    f) ENGINE_FOLDER=${OPTARG} ;;
+    i) CLUSTER_ID=${OPTARG} ;;
+    o) OPS_FOLDER=${OPTARG} ;;
+    p) PASSWORD=${OPTARG} ;;
+    r) ENABLE_RIPPLE_STACK=${OPTARG} ;;
+    *)
+        echo "Invalid argument passed" >&2
+        exit 1
+        ;;
+    esac
+done
+
+if [[ -z $ASSETS_FOLDER || -z $CONFIGS_FOLDER || -z $FORCE_DB_REFRESH || -z $ENGINE_FOLDER || -z $CLUSTER_ID || -z $OPS_FOLDER || -z $PASSWORD || -z $ENABLE_RIPPLE_STACK ]]; then
+    echo "Missing arguments"
+    exit 1
+fi
+
+#================================
+# Set script directory as current
+# Ref: https://stackoverflow.com/a/64168461/2077741
+#================================
+
+SCRIPTS_FOLDER="$(dirname "${BASH_SOURCE[0]}")"
+
+#======
+# Login
+#======
+
+bash "$SCRIPTS_FOLDER/check-login.sh" "$PASSWORD"
+
+checkExitCode
+
+set -e
+
+#=============
+# Verify Node
+#=============
+
+INSTALL_NODE=false
+if node --version >/dev/null; then
+    echo "node is installed"
+
+    NODE_VERSION=$(node --version)
+
+    # Ensure version is greater than 16
+    NODE_MAJOR=${NODE_VERSION:1:2}
+    if [[ $NODE_MAJOR -lt 16 ]]; then
+        echo "node version is outdated $NODE_VERSION"
+        INSTALL_NODE=true
+    fi
+else
+    echo "node is not installed"
+    INSTALL_NODE=true
+fi
+
+if $INSTALL_NODE; then
+    # Ensure NVM is installed
+    if nvm --version >/dev/null; then
+        echo "nvm is installed"
+    else
+        echo "nvm is not installed"
+
+        echo "$PASSWORD" | apt update -y
```

Comment:
Any reason why we are not using it as `echo "$PASSWORD" | sudo -S apt update -y` as I guess `sudo -S` was used to use the echoed password from stdin.

https://discussions.apple.com/thread/5432720?answerId=23353376022#23353376022

Reply:
Macos doesn’t like the sudo commands. So, I was trying to get rid of them. I see this one is inside the VM. Good catch. I’ll change that back.